### ISSUE TITLE
Custom rules for banking not work propertly

### ISSUE THREAD

**Issue by ivsen30** [klingo/ESO-PersonalAssistant#362]
Hi!
First of all, give thank you for your work.

Creation of personal rules for banking and junk dont work properly for my principal character (first in the list, first i create in game) after the latest updates.
Them work fine for the other characters.
I try to erase all data, the add on, reinstall, delete all profiles, create all them again, but nothing works.

In my principal character all the standar rules work properly, but when I create my own rules, them dont work.

Thanks for all.

**Comment by klingo** [comment]
Unfortuantely, I am not able to reproduce your issue. For me the rules are always working fine with any kind of profile or character (also when I deleted all the settings and started from scratch).
Could you please send me your **SavedVariables** files so I can se eif something is wrong there?

You can find them here:
> C:\Users\username\Documents\Elder Scrolls Online\live\SavedVariables\

**Comment by klingo** [comment]
Unfortuantely, I am not able to reproduce your issue. For me the rules are always working fine with any kind of profile or character (also when I deleted all the settings and started from scratch).
Could you please send me your **SavedVariables** files so I can se eif something is wrong there?

You can find them here:
> C:\Users\username\Documents\Elder Scrolls Online\live\SavedVariables\

**Comment by KingShujin** [comment]
If you have items set in the custom rule, you must disable all the categories for the items in your list. 

The addon first initiates the deposit of all selected categories first, then widthraws the items from your custom list.

This causes the addon to deposit items, then withdraw them again (with an incorrect quantity and brief lag spike)

For example, if i set the addon to withdraw 200 potions for me, but also have it set to deposit all potions to my bank it will first deposit all the potions, then withdraw the potions i specified (but would withdraw extra and leave a half stack in my inventory).

The way to fix this would be to prevent the bank from depositing any items in the list in the first place, then later does a check to see if the quanitity is correct, if not then it can withdraw or deposit accordingly.

**Comment by klingo** [comment]
This is working like that on purpose though - but for me it's working fine with the numbers?
Example: I take 586 potions in my backpack and have a rule to only have 100. SO first 586 get deposited and then the custom rule withdraws 100 again:
![image](https://user-images.githubusercontent.com/14639044/116785986-c748f700-aa9c-11eb-8f5e-9194a938ac75.png)

Can you please provide a screenshot of:
- your custom rule
- the chat output that shows the wrong withdrawal

**Comment by ivsen30** [comment]
Hi, sorry, I was traveling for work and couldn't answer before.

Last time I deleted all the personal rules and left only the main rules,
putting all things directly to the bank.

I was setting again the rules to reproduce the problem, but... Now works
right.

Last time, before I wrote to you, I deleted all, even the addon, and
created the rules of the secondary characters.

I created the rules in the first character and didn't work. After that, I
only sell all the potions and crafted things that don't stack because they
were created by myself.
Then put all in the bank, and try to add news rules in the first character
and works properly.

Then I try to take all the potions in my backpack, and try the rule and
works. I try to put everything in the bank and the rules work properly.

I sent the file you asked me.

Not sure what happend, but before I wrote to you I tried all that I think
can be wrong with an update, caches, reinstall all, create the rules again,
and don't work, but now it works properly.

Sorry about worrying you. Excuse me.
